docs/auth.md

```diff
@@ -0,0 +1,192 @@
+# Authentication and authorization
+
+<!-- markdownlint-disable-next-line MD013 -->
+This document describes the authentication and authorization (A&A) of deployed Azure TRE system. The backbone of A&A is [Azure Active Directory (AAD)](https://docs.microsoft.com/en-us/azure/active-directory/fundamentals/active-directory-whatis). It holds the identities of all TRE/workspace users, including administrators, and connects the identities with app registrations defining the privileges per user roles.
+
+## App registrations
+
+App registrations (represented by service principals) define the privileges enabling access to the TRE system (e.g., management API) as well as the workspaces. The two main app registrations, "TRE API" and "TRE Swagger UI", are created by the [`/scripts/aad-app-reg.sh`](../scripts/aad-app-reg.sh) script.
+
+Workspaces rely on app registrations as well, and those are documented under [Workspaces](#workspaces).
+
+### TRE API
+
+TRE API app registration defines the permissions, scopes and app roles for management API users to authenticate and authorize API calls.
+
+#### API permissions - TRE API
+
+| API/permission name | Type | Description | Admin consent required |
+| ------------------- | ---- | ----------- | ---------------------- |
+| Microsoft Graph/Directory.Read.All (`https://graph.microsoft.com/Directory.Read.All`) | Application* | Allows the app to read data in your organization's directory, such as users, groups and apps, without a signed-in user. | Yes |
+| Microsoft Graph/User.Read.All (`https://graph.microsoft.com/User.Read.All`) | Application* | Allows the app to read user profiles without a signed in user. | Yes |
```

Comment:
```suggestion
| API/permission name | Type | Description | Admin consent required | TRE usage |
| ------------------- | ---- | ----------- | ---------------------- | -------- |
| Microsoft Graph/Directory.Read.All (`https://graph.microsoft.com/Directory.Read.All`) | Application* | Allows the app to read data in your organization's directory, such as users, groups and apps, without a signed-in user. | Yes | Management API looks up if a user has permission in a Workspace - based on the application roles assigned. |
| Microsoft Graph/User.Read.All (`https://graph.microsoft.com/User.Read.All`) | Application* | Allows the app to read user profiles without a signed in user. | Yes | .....?  |
```

Reply:
Why do we need the User.Read.All?

Reply:
`User.ReadBasic.All` might be sufficient: https://github.com/microsoft/AzureTRE/issues/384

Reply:
@Lybecker I think that in your suggestion, the TRE usage description you wrote applies to `User.Read.All` instead of `Directory.Read.All` - please confirm. The `Directory.Read.All` permission is quite self-evident so I don't think it needs a separate TRE usage explanation.

Reply:
It was a mistake from my site. Sorry

Reply:
@marrobi do you have a good description on why we need each of these permissions?

Reply:
Not to hand. But if look at the MS graph queries in the auth code can see where the user profile or directory is read - for example to retrieve app registration details, and a user's associated app roles.